### PR TITLE
Add mirror-repo information to all current composer packages

### PR DESCRIPTION
**Pull request by kraftbj**
Builds on top of #18313's framework and defines a mirror repo for all existing Composer packages.

#### Changes proposed in this Pull Request:
* Define mirror repos.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* See 18313.
*

#### Proposed changelog entry for your changes:
* n/a

### CODE REVIEW

**Comment by kraftbj** [comment]
Polled the team in p9dueE-2l9-p2

**Comment by anomiex** [comment]
Ugh, the merge of feature/reorg with various changes and then subsequent changes to master makes this too crazy to try merging master. I'm going to push a rebase.

**Review by anomiex — DISMISSED** [review]
Consensus seems in favor of this.

Asking for a second review of the bits I contributed.

File: .github/files/build-all-projects.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ for project in projects/packages/* projects/plugins/*; do
 		continue
 	fi
 
-	GIT_SLUG=$(jq -r '.extra["mirror-repo"] // ( .name | sub( "^automattic/"; "Automattic/" ) )' composer.json)
+	GIT_SLUG=$(jq -r '.extra["mirror-repo"] // ""' composer.json)
 	if [[ -z "$GIT_SLUG" ]]; then
 		echo "::error::Failed to determine project repo name from composer.json"
 		EXIT=1
```
Comment (kraftbj) [inline .github/files/build-all-projects.sh:37]: Should we skip here, akin to how we do if there is no `composer.json`?

**Comment by kraftbj** [comment]
@anomiex I made one small change to skip a project without a repo set without changing the exit status code. Otherwise, I approve of your work, so if you're happy with this, let's merge it in.

**Review by anomiex — APPROVED** [review]

**Pull request merged**